src/device_handle.rs

```diff
@@ -143,7 +145,27 @@ impl<T: UsbContext> DeviceHandle<T> {
     /// This structure tracks claimed interfaces, and will get out if sync if interfaces are
     /// manipulated externally. Use only libusb endpoint IO functions.
     pub fn as_raw(&self) -> *mut libusb_device_handle {
-        self.handle.as_ptr()
+        // Safety: handle is Some() after initialization
+        match self.handle {
+            Some(it) => it,
```

Comment:
```suggestion
            Some(it) => it.as_ptr(),
```
And remove tail call

And in other place

Reply:
ok